libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -361,6 +367,12 @@ - (void)postPushUnregistration:(SFUserAccount *)user {
         [SFSecurityLockout clearPasscodeState];
     }
     [SFSDKWebViewStateManager removeSession];
+    
+    //restore these id's inorder to enable post logout cleanup of components
+    user.credentials.userId = userId;
```

Comment:
What do we think about changing this interface in a future release?  This seems like a reasonable workaround in the short term—and aligned with the current interface—but in general, it feels like continuing to have an `SFUserAccount` at this point is a violation of the idea that an `SFUserAccount` is an authenticated identity.  @trooper2013 what are your thoughts?

Reply:
By changing the interface, I mean sending something that represents this data in the `kSFNotificationUserDidLogout` notification, but not an `SFUserAccount`.

Reply:
@khawkins agreed. Having partly hydrated SFUserAccount at this point does lend itself to unexpected problems, and violates the definition of an SFUserAccount.  if and when we do provide data, I think it will be necessary to provide some basic supporting functions around it, as in what does one do with these id's?  A small change  for 7.0 perhaps?

Reply:
I think it's probably a small change—and 7.0 is probably the right place, because it's an interface change.  Conceptually, we wouldn't be changing the dynamics of what's happening here, since this user is already unauthenticated at this point.